qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/plot_view.py

```diff
@@ -169,7 +172,7 @@ def clear_figure(self):
 
     def update_figure(self):
         self.toolbar.update()
-        self.update_legend(self.get_axes()[0])
+        self.update_legend(self.get_axes()[0], self.toolbar.get_show_legend_value())
```

Comment:
I feel it's better to keep the call of the function as it was before and use `self.toolbar.get_show_legend_calue()` inside of the `update_legend` function itself.